### PR TITLE
Check if the device is connected

### PR DESCRIPTION
**Pull request by JeremyDezalos**

### CODE REVIEW

File: app/src/main/java/com/github/wnder/NetworkInformation.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import android.net.NetworkInfo;
+
+import static androidx.core.content.ContextCompat.getSystemService;
+
+public class NetworkInformation {
+    public static boolean isNetworkAvailable(Context ctx) {
+        ConnectivityManager manager = (ConnectivityManager) ctx.getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo networkInfo = manager.getActiveNetworkInfo();
+        boolean isAvailable = false;
+        if (networkInfo != null && networkInfo.isConnected()) {
```
Comment (rkuenzi-epfl) [inline app/src/main/java/com/github/wnder/NetworkInformation.java:14]: Just want to make sure: are you sure this verifiy any type of internet connecivity? WiFi and cellular?

File: app/src/main/java/com/github/wnder/NetworkInformation.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.github.wnder;
+
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+
+import static androidx.core.content.ContextCompat.getSystemService;
+
+public class NetworkInformation {
```
Comment (rkuenzi-epfl) [inline app/src/main/java/com/github/wnder/NetworkInformation.java:9]: Once again making a private constructor might be a good idea just to make it clear it's not meant to be intanciated.

**Review by rkuenzi-epfl — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/wnder/NetworkInformation.java (hunk 1 above)
Comment (rkuenzi-epfl) [reply app/src/main/java/com/github/wnder/NetworkInformation.java:24]: Okay thanks

File: app/src/androidTest/java/com/github/wnder/NetworkInformationTest.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+@RunWith(JUnit4.class)
+public class NetworkInformationTest {
+    @Test
+    public void isNetworkAvailableWorksWhenThereIsConnection(){
+        Context context = ApplicationProvider.getApplicationContext();
+
+        assertThat(NetworkInformation.isNetworkAvailable(context), Matchers.is(true));
+    }
+}
```
Comment (leonardvaney) [inline app/src/androidTest/java/com/github/wnder/NetworkInformationTest.java:26]: This test seems incomplete. Did you try to remove the internet connection of the emulator?

Comment (JeremyDezalos) [reply app/src/androidTest/java/com/github/wnder/NetworkInformationTest.java:26]: Yes I tried. At some point in time, you could active the airplane mode but they removed the functionality. I have not found any methods that could do that now.

Comment (leonardvaney) [reply app/src/androidTest/java/com/github/wnder/NetworkInformationTest.java:26]: Ok, that's fine.

**Review by leonardvaney — APPROVED** [review]

**Pull request merged**